cmd/prometheus/main.go

```diff
@@ -1639,6 +1639,11 @@ func (s *readyStorage) Appender(ctx context.Context) storage.Appender {
 
 type notReadyAppender struct{}
 
+// SetHints implements storage.Appender.
+func (n notReadyAppender) SetHints(hints *storage.AppendHints) {
+	panic("unimplemented")
+}
```

Comment:
This shouldn't panic, it should do nothing.

```suggestion
func (n notReadyAppender) SetHints(hints *storage.AppendHints) {}
```